dev/checkstyle-suppressions.xml

```diff
@@ -46,4 +46,12 @@
               files="sql/catalyst/src/main/java/org/apache/spark/sql/streaming/GroupStateTimeout.java"/>
     <suppress checks="MethodName"
               files="sql/core/src/main/java/org/apache/spark/sql/streaming/Trigger.java"/>
+    <suppress checks="RegexpSingleline"
```

Comment:
I think this might suppress many checks? is it difficult to suppress with comments in these files?

Reply:
I found the problem is `RegexpSingleline` can't put into `TreeWalker`, so `SuppressionCommentFilter` rule can't effect on it, `RegexpSinglelineJava` is the right choice. Also spend some time on comment syntax, maybe the usage description can be improved a little, please have a check whether I understand this rule correctly. Done in a4f49ce.